lisa/tools/tar.py

```diff
@@ -10,16 +14,60 @@ class Tar(Tool):
     def command(self) -> str:
         return "tar"
 
-    def extract(self, file: str, dest_dir: str) -> str:
+    def extract(
+        self,
+        file: str,
+        dest_dir: str,
+        strip_components: int = 0,
+    ) -> None:
         # create folder when it doesn't exist
+        assert_that(strip_components).described_as(
+            "--strip-components arg for tar must be int >= 0"
+        ).is_greater_than_or_equal_to(0)
         self.node.execute(f"mkdir -p {dest_dir}", shell=True)
-        result = self.run(
-            f"-xvf {file} -C {dest_dir}", shell=True, force_run=True, sudo=True
-        )
+        tar_cmd = f"-xvf {file} -C {dest_dir}"
+        if strip_components:
+            # optionally strip N top level components from a tar file
+            tar_cmd += f" --strip-components={strip_components}"
+        result = self.run(tar_cmd, shell=True, force_run=True, sudo=True)
         if result.exit_code != 0:
             raise LisaException(
                 f"Failed to extract file to {dest_dir}, {result.stderr}"
             )
-        # Get the folder name of the extracted file
-        source_name = result.stdout.split("/")[0]
-        return source_name
+
+    def list(
+        self, file: str, recursive: bool = True, folders_only: bool = False
+    ) -> List[str]:
+        # return a list of all the files and folders in a tar file
+        result = self.run(
+            f"-tf {file}",
+            shell=True,
+            force_run=True,
+            sudo=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=(
+                f"Could not list items in tar file {file}"
+            ),
+        )
+        content = result.stdout.splitlines()
+
+        if recursive:
+            return content  # get top level components
+        else:
+            files: List[str] = []
+            folders = set()
+            for item in content:
+                if not folders_only:
+                    if "/" not in item:
+                        files += item
+                path_split = item.split("/")
+                if len(path_split) >= 2:
+                    folders.add(path_split[0])
+            return files + list(folders)
+
+    def get_sole_top_level_output_folder(self, file: str) -> str:
```

Comment:
How about a shorter name like `get_root_folder`?

Reply:
That makes sense, I wanted to hint that there's an assumption of a single one, but get_single_root_folder or something is cleaner for sure. Thanks!